### ISSUE TITLE
What is the best way to expose line and column of XML elements and attributes?

### ISSUE THREAD

**Issue by domesticmouse** [renggli/dart-xml#69]
Hey Lukas,

I’m thinking of forking your XML package to retain and expose line and column number ranges for XML attributes and elements. This could be useful for in-place editing and the like.

I’m curious how you would change you parsing strategy to retain these ranges?

**Comment by renggli** [comment]
I believe earlier versions of this packagedid something like this. However, with the push towards a mutable DOM (where any edit would invalidate positions) and supporting incremental parsing (where line and column indexes are ambiguous) I removed it from the public API.

That said, it is relatively easy to collect the position of each DOM element without even changing the existing code. You just need to create a custom subclass of the XML parser that keeps track of the positions. I've hacked together an example: https://github.com/renggli/dart-xml/blob/master/example/xml_pos.dart

```bash
$ dart example/xml_pos.dart example/books.xml
1:1: <?xml version="1.0"?>
2:1: <catalog>...
3:3: <book id="bk101">...
4:5: <author>Gambardella, Matthew</author>
5:5: <title>XML Developer\s Guide</title>
...
```
